glassfish-runner/jsp-platform-tck/pom.xml

```diff
@@ -690,6 +210,19 @@
                             </additionalClasspathElements>
                             <dependenciesToScan>jakarta.tck:${tck.artifactId}</dependenciesToScan>
                             <systemPropertyVariables>
+                                <glassfish.maxHeapSize>2048m</glassfish.maxHeapSize>
+                                <glassfish.postBootCommands>create-file-user --groups staff:mgr:DIRECTOR --passwordfile ${project.build.directory}/../j2ee.pass j2ee
+                                create-file-user --groups guest:OTHERROLE --passwordfile ${project.build.directory}/../javajoe.pass javajoe
+                                set server-config.network-config.protocols.protocol.http-listener-1.http.trace-enabled=true
+                                create-mail-resource --mailuser "user01@james.local" --fromaddress "user01@james.local" --mailhost localhost --debug true mail/Session
+                                create-jms-resource --restype jakarta.jms.Queue --property imqDestinationName=myQueue MY_QUEUE
+                                create-jms-resource --restype jakarta.jms.Topic --property imqDestinationName=myTopic MY_TOPIC
+                                create-jdbc-connection-pool --restype javax.sql.DataSource --datasourceclassname org.apache.derby.jdbc.ClientDataSource --property DatabaseName=derbyDB:serverName=localhost:PortNumber=1527:User=cts1:Password=cts1 --steadypoolsize 32 --maxpoolsize 64 cts-derby-pool
+                                create-jdbc-resource --connectionpoolid cts-derby-pool jdbc/DB1
+                                create-jms-resource --restype jakarta.jms.QueueConnectionFactory jms/QueueConnectionFactory
+                                create-jms-resource --restype jakarta.jms.TopicConnectionFactory jms/TopicConnectionFactory
+                                create-jms-resource --restype jakarta.jms.ConnectionFactory --property name=cFactory jms/ConnectionFactory
+                                create-custom-resource --restype java.net.URL --factoryclass javax.naming.spi.ObjectFactory --property name=myUrl "http\://webServerHost\:webServerPort"</glassfish.postBootCommands>
```

Comment:
The custom resource is created here to resolve below error.

`Caused by: org.omnifaces.arquillian.jersey.server.ContainerException: While Deploying Application: jsp_tagext_resource_taghandler_web --exit_code: FAILURE, message: Error occurred during deployment: Exception while deploying the app [jsp_tagext_resource_taghandler_web] : JNDI lookup failed for the resource: Name: java:module/env/myUrl, Lookup: http://webServerHost:webServerPort, Type: java.net.URL -- Lookup failed for http://webServerHost:webServerPort in SerialContext[myEnv={java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl, java.naming.factory.url.pkgs=com.sun.enterprise.naming}]. Please see server.log for more details. [status: CLIENT_ERROR reason: Bad Request]`

But the command fails with below error : 

```
Executing post boot command using command: [/Users/alwjosep/Softwares/jdk-17.0.2.jdk/Contents/Home/bin/java, -jar, /Users/alwjosep/Documents/jakartaee-tck/glassfish-runner/jsp-platform-tck/target/glassfish8/glassfish/modules/admin-cli.jar, create-custom-resource, --restype, java.net.URL, --factoryclass, javax.naming.spi.ObjectFactory, --property, name=myUrl, "http\://webServerHost\:webServerPort", -t]
remote failure: Unable to create custom resource "http\://webServerHost\:webServerPort". Constraints for this CustomResource configuration have been violated: on property [ jndiName ] violation reason [ Pattern: [^',][^',\]* ], on property [  ] violation reason [ resource name cannot contain ':' character ]
Sep 13, 2024 11:32:50 AM org.omnifaces.arquillian.container.glassfish.managed.GlassFishServerControl start
WARNING: Unable to execute [/Users/alwjosep/Softwares/jdk-17.0.2.jdk/Contents/Home/bin/java, -jar, /Users/alwjosep/Documents/jakartaee-tck/glassfish-runner/jsp-platform-tck/target/glassfish8/glassfish/modules/admin-cli.jar, create-custom-resource, --restype, java.net.URL, --factoryclass, javax.naming.spi.ObjectFactory, --property, name=myUrl, "http\://webServerHost\:webServerPort", -t]
```

@arjantijms @dmatej Would any of you know the correct way to create the above resource in Glassfish or alternate method to resolve the jndi lookup issue.

Reply:
I will take a look this week

Reply:
... or this week, now :-)

Reply:
Would this help? I have found the validation class, but I am not sure why the ':' character is explicitly forbidden in these JNDI names. However I have found another command which passed:
```
asadmin create-jndi-resource --restype java.net.URL --factoryclass javax.naming.spi.ObjectFactory --jndilookupname='http://webServerHost:webServerPort' myUrl
```
Then:
```
asadmin list-jndi-entries
myUrl: org.glassfish.resourcebase.resources.api.ResourceProxy
jdbc: com.sun.enterprise.naming.impl.TransientContext
jms: com.sun.enterprise.naming.impl.TransientContext
UserTransaction: com.sun.enterprise.transaction.startup.TransactionLifecycleService$1
http: com.sun.enterprise.naming.impl.TransientContext
Command list-jndi-entries executed successfully.
```

Reply:
@dmatej I realised that creation of this jndi is not required as it is mentioned in the *sun-web.xml file where the values of webServerHost & webServerPort needs to be replaced with the actual ones.
The tests are now passing with https://github.com/jakartaee/platform-tck/pull/1542. Thanks for your help!